lib/showoff.rb

```diff
@@ -478,6 +479,178 @@ def process_content_for_replacements(content)
+          ],
+          :block => proc { |match_data|
+            inner = process_custom_syntax(match_data[:inner])
+            classes = match_data[:classes].gsub('.',' ')
+            %Q(\n<div markdown="1" class="notes-section #{classes}">\n\n#{inner}\n\n</div>\n\n)
+          },
+        },
+        "code" => {
+          :matchers => [
+            /\s{4}@@@\s(?<languages>.*?)\n\s{4}(?<text>.*?)\n\n/m
+          ],
+          :block => proc { |match_data|
+            css = match_data[:languages].split.collect {|l| "language-#{l.downcase}" }.join(' ')
+            text = match_data[:text].gsub(/^\s{4}/,'')
+            %Q(<pre class="highlight"><code class="#{css}">#{text}</code></pre>)
+          }
+        },
+        "paragraph" => {
+          :matchers => [
+            /^\s+\.(?!break|comment)(?<classes>\S*) (?<inner>.+)$/
+          ],
+          :block => proc { |match_data|
+            inner = process_custom_syntax(match_data[:inner])
+            classes = match_data[:classes].gsub('.', ' ')
+            %Q(\n<p markdown="1" class="#{classes}">\n\n#{inner}\n\n</p>\n\n)
+          },
+        },
+        "pagebreak" => {
+          :matchers => [
+            /~~~PAGEBREAK~~~/
+          ],
+          :block => proc {
+            %Q(<div class="pagebreak">continued...</div>)
+          }
+        },
+        "form" => {
+          :matchers => [
+            /~~~FORM:(?<title>[^~]*)~~~/
+          ],
+          :block => proc { |match_data|
+            %Q(<div class="form wrapper" title="#{match_data[:title]}"></div>)
+          }
+        },
+        # Check for any kind of options
+        "config" => {
+          :matchers => [
+            /~~~CONFIG:(?<setting>.*?)~~~/
+          ],
+          :block => proc { |match_data|
+            settings.showoff_config[match_data[:setting]] if settings.showoff_config.key?(match_data[:setting])
+            # Change setting and return empty string
+            ''
+          }
+        },
+        # Load and replace any file tags
+        "file" => {
+          :matchers => [
+            /~~~FILE:(?<name>[^:~]*):?(?<languages>.*)?~~~/
+          ],
+          :block => proc { |match_data|
+            # make a list of code highlighting classes to include
+            css = match_data[:languages].split.collect {|l| "language-#{l.downcase}" }.join(' ')
+
+            # get the file content and parse out html entities
+            file = File.read(File.join(settings.pres_dir, '_files', match_data[:name])) rescue "Nonexistent file: #{match_data[:name]}"
+            file = "Empty file: #{match_data[:name]}" if file.empty?
+            file = HTMLEntities.new.encode(file) rescue "HTML parsing of #{match_data[:name]} failed"
+
+            %Q(<pre class="highlight"><code class="#{css}">#{file}</code></pre>)
+          }
+        },
+        # Remove comments and breaks from final html
+        "comment_break" => {
```

Comment:
Oh, I see what you mean, the old code actually removes the whole paragraph. Could we just replace it with two newlines instead of an empty p?